C++/source/searching/linear_search.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+    Linear Search
+    -------------
+    A searching algorithm that finds the position of a target value
+    within an array (sorted or otherwise).
+
+    Time complexity
+    --------------
+    O(N), where N is the number of elements in the array
+
+    Space Complexity
+    --------------
+    O(1)
+*/
+
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+int linearSearch(const int value, const vector<int> &vect) {
+
+    // Search all the elements in the array
+    for(i = 0; i < vect.length(); i++) {
```

Comment:
* `i` should be declared as a `size_t`
* `length()` does not exist for a `vector`. The right function to use is [`size()`](http://en.cppreference.com/w/cpp/container/vector/size).